### PR TITLE
github CI

### PR DESCRIPTION
**Pull request by konradybcio**
need some help with fedora..

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -9,24 +9,30 @@ project('qrtr',
...
 prefix = get_option('prefix')
-systemd = dependency('systemd')
 
-systemd_system_unit_dir = get_option('systemd-unit-prefix')
-if systemd_system_unit_dir == ''
-        systemd_system_unit_dir = systemd.get_variable(
-                pkgconfig : 'systemdsystemunitdir',
+install_systemd_unit = get_option('systemd-service')
+if install_systemd_unit.auto() or install_systemd_unit.enabled()
+        systemd = dependency('systemd')
```
Comment (lumag) [inline meson.build]: Please use meson way of doing things. Add `required: false` and then use `if systemd.found()`

**Review by lumag — CHANGES_REQUESTED** [review]

File: meson.build [hunk 2]
```diff
@@ -9,24 +9,29 @@ project('qrtr',
...
-                pkgconfig : 'systemdsystemunitdir',
+install_systemd_unit = get_option('systemd-service')
+systemd = dependency('systemd', required : install_systemd_unit)
+if systemd.found()
+        systemd_system_unit_dir = get_option('systemd-unit-prefix')
+        if systemd_system_unit_dir == ''
+                systemd_system_unit_dir = systemd.get_variable(
+                        pkgconfig : 'systemdsystemunitdir',
                 pkgconfig_define: ['prefix', prefix])
+        endif
```
Comment (lumag) [inline meson.build:24]: I'd simply warn the user about skipping installation of systemd units instead of failing here.

File: meson.build [hunk 3]
```diff
@@ -3,7 +3,6 @@
 project('qrtr',
         'c',
         license : [ 'BSD-3-Clause'],
-        meson_version : '>= 0.64.0', # for feature user options
```
Comment (superna9999) [inline meson.build:6]: well it's useful because you use features that are not available on older meson versions

Comment (konradybcio) [reply meson.build:6]: idk.. we have old debian in CI and it goes through

File: meson.build (hunk 2 above)
Comment (konradybcio) [reply meson.build:24]: there's no failing involved

Comment (lumag) [reply meson.build:24]: It does. Fedora builds fail.

Comment (konradybcio) [reply meson.build:24]: aaahh, you mean like that.. okay..

Comment (konradybcio) [reply meson.build:24]: yeah hm I have no idea how to solve that and I'm not sure it makes sense if the user doesn't explicitly disable `-Dsystemd-service`

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+        uname -a
+        echo "############################################"
+        cat /proc/cmdline
+        echo "############################################"
+        printenv
+
+    - name: Disable systemd
+      if: ${{ matrix.nosystemd == true }}
+      run: |
+        touch .nosystemd
```
Comment (lumag) [inline .github/workflows/ci.yml]: I'd say, drop this completely. The build system should be able to detect if systemd is present or not.

**Review by lumag — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 4 above)
Comment (lumag) [reply .github/workflows/ci.yml]: *now should be able*

**Review by lumag — APPROVED** [review]

**Pull request merged**